### DISCUSSION TITLE
Recodings - time display in list

### DISCUSSION THREAD

**Discussion by petricf** [Motion-Project/motionplus#101] [General]
Under "Recordings" -> "\<camera>" there is a list of recordings in tabular form.

The column "time" seems to be the start time of the recording. I expect either 12 hour (with AM/PM) format or the 24 hour format. But it seems to be displayed in the 12 hour format (w/o AM/PM suffix).
This way you can not see whether to recording was made in the morning or in the evening.

The file names (*.mkv) seems to be ok. The time stamps used to display (column movie_tmc) in the database looks wrong. The column movie_tml looks better but did not exactly match the file names.

Or did i miss something ?

**Comment by Mr-Dave** [comment]
Not sure what would be causing all the things you are describing.  

The times won't match exactly between the file name and database.  The database creates the timestamp as it is executed.  

As for the am/pm, my display includes it.  e.g. `pls79_20231005053803.mkv | 5.8MB | 20231005 | 05:39AM | 233 | 3 | 149 | 87`  

For the  movie times in the database, they are created [here](https://github.com/Motion-Project/motionplus/blob/44224b28e987d3bf2018f79274afb6f90b333455/src/dbse.cpp#L1349) and as you can see, they are all formatting the exact same date just into different formats.   The "movie_tmc" is 12 hour minute with am/pm while the "movie_tml" is 24 hour minute second

**Reply by petricf** [reply]
I see following in the list:

    Name | Size | Date | time | diff_avg | sdev_min | sdev_max | sdev_avg |  
    Garden_101-01-20231003211126.mkv | 16,0MB | 20231003 | 09:13 | 2796 | 121 | 168 | 152 |  
    Garden_101-02-20231003211439.mkv | 7,6MB | 20231003 | 09:15 | 6430 | 65 | 129 | 102 |  

As you can see there is no AM/PM in the time column. The file name indicates there should be a PM attached.

The contents of the database with the entries above:

    MariaDB [motionplus]> select * from motionplus limit 2;
    +-----------+-----------+--------------------------------------+---------------------+----------------------------------------------------------+----------+-----------+-----------+-----------+----------+----------+----------+----------+
    | record_id | device_id | movie_nm                             | movie_dir           | full_nm                                                  | movie_sz | movie_dtl | movie_tmc | movie_tml | diff_avg | sdev_min | sdev_max | sdev_avg |
    +-----------+-----------+--------------------------------------+---------------------+----------------------------------------------------------+----------+-----------+-----------+-----------+----------+----------+----------+----------+
    |       416 |       101 | Garden_101-01-20231003211126.mkv | /footage/garden | /footage/garden/Garden_101-01-20231003211126.mkv | 15990784 |  20231003 | 09:13     | 21:13:09  |     2796 |      121 |      168 |      152 |
    |       418 |       101 | Garden_101-02-20231003211439.mkv | /footage/garden | /footage/garden/Garden_101-02-20231003211439.mkv |  7602176 |  20231003 | 09:15     | 21:15:51  |     6430 |       65 |      129 |      102 |
    +-----------+-----------+--------------------------------------+---------------------+----------------------------------------------------------+----------+-----------+-----------+-----------+----------+----------+----------+----------+

**Reply by Mr-Dave** [reply]
What is your locale?

**Reply by petricf** [reply]
It is a standard debian 12 installation (w/o GUI) localized to german.

    root@motion-server:/etc/motionplus# locale
    LANG=de_DE.UTF-8
    LANGUAGE=
    LC_CTYPE="de_DE.UTF-8"
    LC_NUMERIC="de_DE.UTF-8"
    LC_TIME="de_DE.UTF-8"
    LC_COLLATE=C
    LC_MONETARY="de_DE.UTF-8"
    LC_MESSAGES="de_DE.UTF-8"
    LC_PAPER="de_DE.UTF-8"
    LC_NAME="de_DE.UTF-8"
    LC_ADDRESS="de_DE.UTF-8"
    LC_TELEPHONE="de_DE.UTF-8"
    LC_MEASUREMENT="de_DE.UTF-8"
    LC_IDENTIFICATION="de_DE.UTF-8"
    LC_ALL=

**Reply by Mr-Dave** [reply]
That is what I was guessing.  German does not have a AM/PM.  As such, the strftime function localizes the AM/PM to nothing.

**Reply by petricf** [reply]
In Germany we have 24 hour time display - 00:00 to 23:59

**Comment by funix83** [comment]
I have exactly the same problem (in France, we also have 24 hour time display), I don't know if the video was recorded during the day or the night.
Do you plan to correct this issue ?
